### PR TITLE
java: async-profiler: Set default mode to "cpu"

### PR DESCRIPTION
**Pull request by Jongy**
## Description
It's time :)

We now trust the CPU mode enough to set it by default.

## How Has This Been Tested?
CI, and in some of our deployments.

* I tested the auto selection logic by running gProfiler from a shell started by `sudo capsh --drop=CAP_SYS_ADMIN`. It selected `itimer`.
* Also, `test_executable_not_privileged` tests it, because that test fails if `cpu` is used.
* And I modified `test_java_async_profiler_cpu_mode` to use `auto`. So if `itimer` is picked, test will fail because it wants to see kernel stacks.

### CODE REVIEW

**Review by liorgorb — DISMISSED** [review]

**Comment by Jongy** [comment]
@liorgorb ready for CR

**Comment by Jongy** [comment]
Now it's passing... How nice

**Review by adi-benz — CHANGES_REQUESTED** [review]
I can't really comment on the core essence of this PR, I don't have a good understanding on the differences between `itimer` and `cpu` (other than docs and a little bit of code reading), and I don't know what's the problems `cpu` had in the past that now we trust it.

So as far as reviewing the simple code that chooses the mode - no problem  
But I don't have any meaningful comments about `cpu` mode vs `itimer`, so maybe we want someone that knows it to take a look?

**Comment by Jongy** [comment]
> I can't really comment on the core essence of this PR, I don't have a good understanding on the differences between itimer and cpu (other than docs and a little bit of code reading), and I don't know what's the problems cpu had in the past that now we trust it.

Generally: CPU is based on `perf_events` which we have concluded to be more accurate than `itimer`, so we want it to be used when available.

We use CPU mode with `fdtransfer` which is a mechanism that [I've added to async-profiler](https://github.com/jvm-profiling-tools/async-profiler/pull/411) a while ago. It was a new feature in AP and we wanted to have some "run time" before we enable it by default. We've had a while with it and now we deem it ready.

File: tests/utils.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ def run_privileged_container(
 
 
 def _no_errors(logs: str):
+    print(logs)  # will be displayed if the test fails
```
[the remark this answers is not in the document]
Comment (Jongy) [reply tests/utils.py:62]: pytest doesn't show the stdout of passing tests, only upon failure, so it's a common pattern

File: gprofiler/utils.py [hunk 2]
```diff
@@ -488,3 +488,26 @@ def get_kernel_release() -> Tuple[int, int]:
     """Return Linux kernel version as (major, minor) tuple."""
     major_str, minor_str = os.uname().release.split(".", maxsplit=2)[:2]
     return int(major_str), int(minor_str)
+
+
+def perf_path() -> str:
```
[the remark this answers is not in the document]
Comment (Jongy) [reply gprofiler/utils.py:495]: Well, we have `perf.py` for the profiler but it doesn't belong there now I suppose. Time to create a utils directory... And slowly slowly split up this large file.

**Comment by Jongy** [comment]
> But I don't have any meaningful comments about cpu mode vs itimer, so maybe we want someone that knows it to take a look?

Was approved by @liorgorb and @YoniKF as well.

File: tests/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (adi-benz) [reply tests/utils.py:62]: Running pytest with `-s` makes it show stdout, which in my experience is quite useful for debugging and such.  
Also, PyCharm shows stdout by default when clicking on each test.

I don't see any downsides to printing the logs using the assert, it's also simpler (you won't feel the need to document `# will be displayed if the test fails` on an assert)

**Comment by adi-benz** [comment]
> > But I don't have any meaningful comments about cpu mode vs itimer, so maybe we want someone that knows it to take a look?
> 
> Was approved by @liorgorb and @YoniKF as well.

Cool 👍

File: tests/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jongy) [reply tests/utils.py:62]: > Running pytest with -s makes it show stdout, which in my experience is quite useful for debugging and such.
Also, PyCharm shows stdout by default when clicking on each test.

Hmm weird. I'm using `pytest -s` by default but I see the outputs only upon failure.

Will move to the assert, k

[the remark this answers is not in the document]
Comment (adi-benz) [reply tests/utils.py]: I tested it to make sure

image

**Review by adi-benz — APPROVED** [review]

**Pull request merged**